src/DependencyInjection/Configuration.php

```diff
@@ -89,6 +89,23 @@ public function getConfigTreeBuilder()
                     ->info('Enable the global locale switcher services.')
                     ->defaultFalse()
                 ->end()
+                // NEXT_MAJOR: Fix locale to country flag mapping OR remove country flags entirely
+                ->booleanNode('locale_switcher_show_country_flags')
+                    ->info('Whether the language switcher should show languages as flags')
+                    ->defaultTrue()
+                ->end()
+            ->end()
+            ->beforeNormalization()
+                ->ifTrue(function ($v) {return !isset($v['locale_switcher_show_country_flags']) || true === $v['locale_switcher_show_country_flags']; })
+                ->then(function ($v) {
+                    @trigger_error(sprintf(
+                        'Showing the country flags is deprecated. The flags will be removed in the next major version unless the maintainers of this bundle fix the locale to country mapping. Please set "%s" to false to avoid this message.',
```

Comment:
I don't know about you @dbu but I don't plan to maintain a long list of hacks.
```suggestion
                        'Showing the country flags is deprecated. The flags will be removed in the next major version. Please set "%s" to false to avoid this message.',
```